src/ducks/sep8Send.ts

```diff
@@ -0,0 +1,165 @@
+          approval_server: approvalServer,
+          regulated: isRegulated,
+        } = currency;
+
+        if (!approvalCriteria) {
+          throw new Error(
+            "The anchor toml file does not contain an approval criteria.",
+          );
+        }
+
+        if (!approvalServer) {
+          throw new Error(
+            "The anchor toml file does not contain an approval server.",
+          );
+        }
+
+        // this is unlikely
+        if (!isRegulated) {
+          throw new Error(
+            'The anchor toml file does not specify this asset as "regulated".',
+          );
+        }
+
+        return {
+          approvalCriteria,
+          approvalServer,
+          assetCode,
+          assetIssuer,
+          homeDomain,
+          isRegulated,
+        };
+      }
+    }
+
+    throw new Error("Couldn't find the desired asset in the anchor toml file.");
+  } catch (error) {
+    const errorString = getErrorMessage(error);
+    log.error({
+      title: errorString,
+    });
+    return rejectWithValue({ errorString });
+  }
+});
+
+export const sep8SendPaymentAction = createAsyncThunk<
+  Horizon.TransactionResponse,
+  Sep8PaymentTransactionParams,
+  { rejectValue: RejectMessage; state: RootState }
+>(
+  "sep8Send/sep8SendPaymentAction",
+  async (params, { rejectWithValue, getState }) => {
+    const { pubnet: isPubnet, secretKey } = settingsSelector(getState());
+
+    try {
+      const result = await submitSEP8PaymentTransaction({
+        params,
+        secretKey,
+        isPubnet,
+      });
+      return result;
+    } catch (error) {
+      const errorString = getErrorString(error);
+      log.error({ title: errorString });
+      return rejectWithValue({ errorString });
+    }
+  },
+);
+
+const initialState: Sep8SendInitialState = {
+  data: {
+    approvalCriteria: "",
+    approvalServer: "",
+    assetCode: "",
+    assetIssuer: "",
+    homeDomain: "",
+    isRegulated: false,
+  },
+  errorString: undefined,
+  status: undefined,
+};
+
+const sep8SendSlice = createSlice({
+  name: "sep8Send",
+  initialState,
+  reducers: {
+    resetSep8SendAction: () => initialState,
+  },
+  extraReducers: (builder) => {
+    builder.addCase(initiateSep8SendAction.pending, (state = initialState) => {
+      state.status = ActionStatus.PENDING;
+    });
+    builder.addCase(initiateSep8SendAction.fulfilled, (state, action) => {
+      state.data = { ...state.data, ...action.payload };
+      state.status = ActionStatus.CAN_PROCEED;
+    });
+    builder.addCase(initiateSep8SendAction.rejected, (state, action) => {
+      state.errorString = action.payload?.errorString;
+      state.status = ActionStatus.ERROR;
+    });
+
+    builder.addCase(sep8SendPaymentAction.pending, (state) => {
+      state.errorString = undefined;
+      state.status = ActionStatus.PENDING;
+    });
+    builder.addCase(sep8SendPaymentAction.fulfilled, (state) => {
+      // state.data = action.payload;
```

Comment:
can we remove this?

Reply:
Definitely, thanks for spotting that! https://github.com/stellar/stellar-demo-wallet/pull/155/commits/9b0d15c1f8fd1a5528cbd2b98b453a8da786efab